src/build/env.py

```diff
@@ -97,6 +97,19 @@ def _restore_env(self):  # type: () -> None
     def _get_env_path(self, path):  # type: (str) -> Optional[str]
         return sysconfig.get_path(path, vars=self._env_vars)
 
+    def _symlink_file(self, path, new_subpath):  # type: (str, str) -> None
```

Comment:
A better name for this would be _symlink_or_copy, because it falls back to copy.

Reply:
The intent here is to symlink, copy is just a last resort fallback. I stand by the original naming but I can change if you feel strongly about it.

Reply:
The thing is it can happen that it doesn't symlink in which case the name is missleading, so let's address that.

Reply:
What about `place_file`? And similarly rename `symlink_relative` to `place_path_relative`?

Reply:
I feel like `symlink_or_copy` is the best, most descriptive name for this, to be honest. It's a little more verbose, but it's immediately obvious what the function would do, and I think it best serves the two important audiences: people who know that symlinks are not universally available and people who don't know that.

For people who know that symlinks aren't available on all platforms:

- `symlink_or_copy` makes it pretty obvious that the intention is to symlink, but that it falls back to copying (why else would there be any doubt as to which one you're doing)
- If it's just named `symlink`, I am suddenly in doubt as to whether it throws deliberately, fails or falls back to something else.

For people who don't know that symlinks are available on all platforms:

 - `symlink_or_copy` alerts them that they may not understand exactly what's going on in the function (why is there any doubt?), and that in some cases the file may be _copied_ (helping them debug their code more easily as they try and figure out _why_ it would sometimes be copied).
 - If it's named `symlink`, they may not realize that there are edge cases where symlinking won't take place — they won't realize that they don't quite understand what's going on here.

I understand the desire to be more succinct, but I think `place` is also bad. To me it brings up uncertainty as to whether you are creating these files, moving them or copying them. I wouldn't think that `place` describes symlinking at all, much less that symlink is the _more likely_ behavior.